WordPress/Classes/Utility/ImmuTable.swift

```diff
@@ -0,0 +1,206 @@
+    case Nib(UINib)
+    case Class(AnyClass)
+}
+
+public protocol CellRegistrator {
+    func register(registrable: CellRegistrable, cellReuseIdentifier: String)
+}
+
+
+extension UITableView: CellRegistrator {
+    public func register(registrable: CellRegistrable, cellReuseIdentifier: String) {
+        switch registrable {
+        case .Nib(let nib):
+            registerNib(nib, forCellReuseIdentifier: cellReuseIdentifier)
+        case .Class(let cellClass):
+            registerClass(cellClass, forCellReuseIdentifier: cellReuseIdentifier)
+        }
+    }
+}
+
+public struct ImmuTable {
+    public let sections: [ImmuTableSection]
+
+    public func rowAtIndexPath(indexPath: NSIndexPath) -> ImmuTableRow {
+        return sections[indexPath.section].rows[indexPath.row]
+    }
+
+    public static func registerRows(rows: [ImmuTableRow.Type], tableView: CellRegistrator) {
+        let registrables = rows.reduce([:]) {
+            (var classes, row) -> [String: CellRegistrable] in
+
+            classes[row.reusableIdentifier] = row.registrable
+            return classes
+        }
+        for (identifier, registrable) in registrables {
+            tableView.register(registrable, cellReuseIdentifier: identifier)
+        }
+    }
+}
+
+public class ImmuTableDataSource: NSObject, UITableViewDataSource {
+    var viewModel: ImmuTable
+    var configureCell: ((UITableViewCell) -> Void)?
+
+    init(viewModel: ImmuTable) {
+        self.viewModel = viewModel
+    }
+
+    public func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return viewModel.sections.count
+    }
+
+    public func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.sections[section].rows.count
+    }
+
+    public func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        let cell = tableView.dequeueReusableCellWithIdentifier(row.reusableIdentifier, forIndexPath: indexPath)
+
+        row.configureCell(cell)
+
+        configureCell?(cell)
+
+        return cell
+    }
+}
+
+public class ImmuTableDelegate: NSObject, UITableViewDelegate {
+    var viewModel: ImmuTable
+
+    init(viewModel: ImmuTable) {
+        self.viewModel = viewModel
+    }
+
+    public func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        if let action = row.action {
+            action(row)
+        }
+    }
+
+    public func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        if let customHeight = row.dynamicType.customHeight {
+            return CGFloat(customHeight)
+        }
+        return tableView.rowHeight
+    }
+}
+
+public struct ImmuTableViewHandler {
+    let target: UITableViewController
```

Comment:
I think it should be `unowned`, and yeah I could probably use a `TableViewController` protocol, but I wonder if that just complicates things, given that we plan to use it with `UITableViewController`